tests/kafkatest/tests/streams/streams_eos_test.py

```diff
@@ -110,6 +109,7 @@ def run_failure_and_recovery(self, processor1, processor2, processor3, verifier)
         self.abort_streams(processor1, processor3, processor2)
         self.stop_streams2(processor1, processor3)
         self.stop_streams(processor1)
+        processor1.clean_node_enabled = True
 
```

Comment:
We do similar here because that was actually the issue in the system tests. Tests `test_failure_and_recovery()` and  `test_failure_and_recovery_complex()` were executed before `test_rebalance_simple` and `test_rebalance_complex` and they were scheduled on the same worker node, respectively. Thus, the latter tests read the local state of the former tests instead of starting with an empty state.